.github/workflows/doc_build.yml

```diff
@@ -0,0 +1,75 @@
+name: Doc Build
+
+on:
+  workflow_run:
+    types:
+      - completed
+    workflows:
+      - Linux CI (pytorch-linux-xenial-py3.6-gcc5.4)
+  pull_request:
+
+jobs:
+  calculate-docker-image:
+    runs-on: ubuntu-18.04
+    outputs:
+      docker_image: ${{ steps.calculate-tag.outputs.docker_image }}
+    steps:
+      - name: Checkout PyTorch
+        uses: actions/checkout@v2
+      - name: Calculate docker image tag
+        id: calculate-tag
+        run: |
+          DOCKER_TAG=$(git rev-parse HEAD:.circleci/docker)
+          echo "::set-output name=docker_image::${DOCKER_IMAGE_BASE}:${DOCKER_TAG}"
+  pytorch_python_doc_build:
+    runs-on: ubuntu-18.04
+    needs: calculate-docker-image
+    env:
+      DOCKER_IMAGE: ${{ needs.calculate-docker-image.outputs.docker_image }}
+    steps:
+      - name: Chown workspace
+        run: |
+          # Ensure the working directory gets chowned back to the current user
+          docker run --rm -v "$(pwd)":/v -w /v alpine chown -R "$(id -u):$(id -g)" .
+      - name: Checkout PyTorch
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # deep clone, to allow sharding to use git rev-list
+          submodules: recursive
+      - name: Log in to ECR
+        run: |
+          aws ecr get-login --no-include-email --region us-east-1 > /tmp/ecr-login.sh
+          bash /tmp/ecr-login.sh
+          rm /tmp/ecr-login.sh
+      - name: Pull docker image
+        run: |
+          docker pull "${DOCKER_IMAGE}"
+      - name: Preserve github env variables for use in docker
+        run: |
+          env | grep '^GITHUB' > "/tmp/github_env_${GITHUB_RUN_ID}"
+      - name: Build Python Doc
```

Comment:
I think it might be better to make this a part of the regular linux CI workflow template and just have it gated like so:
```
if: {{ env.BUILD_ENVIRONMENT == "pytorch-linux-xenial-py3.6-gcc5.4"
```

It'd make it easier to grab the artifacts etc.

Reply:
makes sense, the artifact upload/download looks like can only be shared within the same workflow.